src/sdk/pynni/nni/nas_utils.py

```diff
@@ -205,4 +233,106 @@ def training_update(nas_mode, tf=None, session=None, loss=None, feed_dict=None):
     if nas_mode == 'darts_mode':
         darts_training(tf, session, loss, feed_dict)
     elif nas_mode == 'enas_mode':
-        reload_tensorflow_variables(tf, session)
\ No newline at end of file
+        reload_tensorflow_variables(tf, session)
+
+
+def _get_layer_and_inputs_from_tuner(mutable_id, mutable_layer_id, optional_inputs):
+    # optional_inputs should be name(key)s of the optional inputs
+    try:
+        mutable_block = trial.get_current_parameter(mutable_id)
+
+        # Great! There is a NAS tuner
+        chosen_layer = mutable_block[mutable_layer_id]["chosen_layer"]
+        chosen_inputs = mutable_block[mutable_layer_id]["chosen_inputs"]
+    except KeyError:
+        # Try to find converted NAS parameters
+        params = trial.get_current_parameter()
+        expected_prefix = _construct_general_key(mutable_id, mutable_layer_id)
+        chosen_layer = params[expected_prefix + "/layer_choice"]
+
+        # find how many to choose
+        optional_input_size = int(params[expected_prefix + "/optional_input_size"])  # convert uniform to randint
+
+        # find who to choose, can duplicate
+        optional_input_state = params[expected_prefix + "/optional_input_chosen_state"]
+        chosen_inputs = []
+        # make sure dict -> list produce stable result by sorting
+        optional_inputs_keys = sorted(optional_inputs)
+        for i in range(optional_input_size):
+            chosen_inputs.append(optional_inputs_keys[optional_input_state % len(optional_inputs)])
+            optional_input_state //= len(optional_inputs)
+
+    _logger.info("%s_%s: layer: %s, optional inputs: %s" % (mutable_id, mutable_layer_id,
+                                                            chosen_layer, chosen_inputs))
+    return chosen_layer, chosen_inputs
+
+
+def convert_nas_search_space(search_space):
```

Comment:
do we have any unit test for this function?

Reply:
@ultmaster do you write unittest for it?

Reply:
I didn't because I didn't know where I should put my unittest. Should it be under `sdk/pynni/nni/tests`? In that case, this whole `nas_utils.py` does not have unittest...

I've done some integrated test (by modifying examples in NAS, using TPE); again I didn't check it in because didn't know where I should put it.

Reply:
@ultmaster would be better to write unittest for it. could be another pr.